src/arch/x86_64.rs

```diff
@@ -190,14 +210,94 @@ pub fn oops(
     let _ = vga.write_str("\n  it will never be safe to turn off your computer.");
 
     #[cfg(test)]
-    qemu_exit(QemuExitCode::Failed);
+    {
+        if let Some(test) = ptr::NonNull::new(CURRENT_TEST.load(Ordering::Acquire)) {
+            if let Some(com1) = serial::com1() {
```

Comment:
A bit of me wonders if we should have a second `-serial` command in the test runner which we use for test progress signalling etc, so we don't need to worry as much about parsing it out of main serial output.

Would make it harder to split up serial output to be grouped by which test failed, I suppose.

Reply:
i thought about that --- this feels easier to just get it working, but we can keep messing with it.